lib/sgr.js

```diff
@@ -0,0 +1,115 @@
+'use strict';
+/* CSI - control sequence introducer */
+/* SGR - set graphic rendition */
+
+var assign   = require('es5-ext/object/assign')
+  , includes = require('es5-ext/string/#/contains');
+
+var CSI = '\x1b[';
+
+var sgr = function (code) {
+	return CSI + code + 'm';
+};
+
+sgr.CSI = CSI;
+
+var mods = assign({
+	// Style
+	bold:      { _bold: [1, 22] },
+	italic:    { _italic: [3, 23] },
+	underline: { _underline: [4, 24] },
+	blink:     { _blink: [5, 25] },
+	inverse:   { _inverse: [7, 27] },
+	strike:    { _strike: [9, 29] }
+
+	// Color
+}, ['black', 'red', 'green', 'yellow', 'blue', 'magenta', 'cyan', 'white']
+	.reduce(function (obj, color, index) {
+		// foreground
+		obj[color] = { _fg: [30 + index, 39] };
+		obj[color + 'Bright'] = { _fg: [90 + index, 39] };
+
+		// background
+		obj['bg' + color[0].toUpperCase() + color.slice(1)] = { _bg: [40 + index, 49] };
+		obj['bg' + color[0].toUpperCase() + color.slice(1) + 'Bright'] = { _bg: [100 + index, 49] };
+
+		return obj;
+	}, {}));
+
+sgr.mods = mods;
+
+(function () {
```

Comment:
This is much more style issue, I want to distinct it from another part of this module. I personally decompose my code very deep, creating many small files, however, I don't usually do this in PRs, but I wanted to show this part of module is like another module, so I've put it in closure and put some local `require`s at the top of it. I like to place `require`s close (as possible) to the place where they used (so it is not at the top of the module in 100% cases).

I'm not persist on this. I can unwrap it.